File: VB/InlinePictures/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
#Region "#using"
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.API.Native
' ...
#End Region ' #using
Imports System.Reflection
Imports System.IO
Imports System.Resources

Namespace InlinePictures
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			richEditControl1.LoadDocument("Texts\Lorem.rtf", DocumentFormat.Rtf)
		End Sub

Private Sub btnFromImage_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnFromImage.Click
'#Region "#fromimage"

Dim pos As DocumentPosition = richEditControl1.Document.CaretPosition
Dim imageToInsert As Image = imageCollection1.Images(3)
richEditControl1.Document.InsertImage(pos, imageToInsert)

'#End Region ' #fromimage
End Sub

Private Sub btnFromFile_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnFromFile.Click
'#Region "#fromfile"
Dim pos As DocumentPosition = richEditControl1.Document.CaretPosition

Dim dialog As New OpenFileDialog()
dialog.Filter = "Image Files|*.jpg;*.gif;*.bmp;*.png;*.jpeg|All Files|*.*"
dialog.FilterIndex = 1
dialog.RestoreDirectory = True

If dialog.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
	Dim imageToInsert As String = dialog.FileName
	richEditControl1.Document.InsertImage(pos, DocumentImageSource.FromFile(imageToInsert))
End If
'#End Region ' #fromfile

End Sub

Private Sub btnFromStream_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnFromStream.Click
'#Region "#fromstream"
Dim pos As DocumentPosition = richEditControl1.Document.CaretPosition
Dim imageToInsert As String = "information.png"
Dim a As System.Reflection.Assembly = System.Reflection.Assembly.GetExecutingAssembly()

Dim imageStream As Stream = a.GetManifestResourceStream("Resources." & imageToInsert)
richEditControl1.Document.InsertImage(pos, DocumentImageSource.FromStream(imageStream))
'#End Region ' #fromstream
End Sub

Private Sub btnScaleDown_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnScaleDown.Click
'#Region "#scaledown"
richEditControl1.Document.Unit = DevExpress.XtraRichEdit.DocumentUnit.Millimeter
Dim images As DocumentImageCollection = richEditControl1.Document.GetImages(richEditControl1.Document.Range)

For i As Integer = 0 To images.Count - 1
	If images(i).Size.Width > 50 Then
		images(i).ScaleX /= 2
		images(i).ScaleY /= 2
	End If
Next i
'#End Region ' #scaledown
End Sub

Private Sub btnFromUri_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnFromUri.Click
'#Region "#fromuri"
Dim pos As DocumentPosition = richEditControl1.Document.CaretPosition
Dim imageUri As String = "http://www.devexpress.com/App_Themes/DX/Common/footer_logo.gif"
richEditControl1.Document.InsertImage(pos, DocumentImageSource.FromUri(imageUri,richEditControl1))
'#End Region ' #fromuri
End Sub
	End Class
End Namespace

File: CS/InlinePictures/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
#region #using
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.API.Native;
// ...
#endregion #using
using System.Reflection;
using System.IO;
using System.Resources;

namespace InlinePictures {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            richEditControl1.LoadDocument("Texts\\Lorem.rtf", DocumentFormat.Rtf);
        }

private void btnFromImage_Click(object sender, EventArgs e) {
#region #fromimage

DocumentPosition pos = richEditControl1.Document.CaretPosition;
Image imageToInsert = imageCollection1.Images[3];
richEditControl1.Document.Images.Insert(pos, imageToInsert);

#endregion #fromimage
}

private void btnFromFile_Click(object sender, EventArgs e) {
#region #fromfile
DocumentPosition pos = richEditControl1.Document.CaretPosition;

OpenFileDialog dialog = new OpenFileDialog();
dialog.Filter = "Image Files|*.jpg;*.gif;*.bmp;*.png;*.jpeg|All Files|*.*";
dialog.FilterIndex = 1;
dialog.RestoreDirectory = true;

if(dialog.ShowDialog() == DialogResult.OK) {
    string imageToInsert = dialog.FileName;
    richEditControl1.Document.Images.Insert(pos, 
        DocumentImageSource.FromFile(imageToInsert));
}
#endregion #fromfile

}

private void btnFromStream_Click(object sender, EventArgs e) {
#region #fromstream
DocumentPosition pos = richEditControl1.Document.CaretPosition;
string imageToInsert = "information.png";
Assembly a = Assembly.GetExecutingAssembly();

Stream imageStream = a.GetManifestResourceStream("InlinePictures.Resources." + imageToInsert);
richEditControl1.Document.Images.Insert(pos, DocumentImageSource.FromStream(imageStream));
#endregion #fromstream
}

private void btnScaleDown_Click(object sender, EventArgs e) {
#region #scaledown
richEditControl1.Document.Unit = 
    DevExpress.Office.DocumentUnit.Millimeter;
ReadOnlyDocumentImageCollection images =
    richEditControl1.Document.Images.Get(richEditControl1.Document.Range);

for(int i = 0; i < images.Count; i++) {
    if(images[i].Size.Width > 50) {
        images[i].ScaleX /= 2;
        images[i].ScaleY /= 2;
    }
}
#endregion #scaledown
}

private void btnFromUri_Click(object sender, EventArgs e) {
#region #fromuri
DocumentPosition pos = richEditControl1.Document.CaretPosition;
string imageUri = "http://www.devexpress.com/App_Themes/DX/Common/footer_logo.gif";
richEditControl1.Document.Images.Insert(pos, DocumentImageSource.FromUri(imageUri, richEditControl1));
#endregion #fromuri
}
    }
}